### PR TITLE
XT-1487 - Fix versioning for pypi packages

### PR DESCRIPTION
**Pull request by derekgengenbacher-wf**
#### Description:
- Make the versioning for the pypi package creation more robust

#### Testing:
- CI

**review**:
@Workiva/xt

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by austinmatherne-wk — DISMISSED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -13,14 +13,10 @@ RUN pip install -r requirements-dev.txt
...
-# with the tagged version number from GIT_TAG or `0.0.0` if GIT_TAG is not set
-ARG VERSION=${GIT_TAG:-0.0.0}
-RUN echo "Version = $VERSION"
-# Update line 6 in setup.py
-RUN sed -i "6 s/0.0.0/$VERSION/" setup.py
+# The following command replaces the @VERSION@ string in setup.py with the tagged version number from GIT_TAG
+RUN sed -i s/@VERSION@/$GIT_TAG/ setup.py
 # Update line 3 in package.json
-RUN sed -i "3 s/0.0.0/$VERSION/" package.json
+RUN sed -i "3 s/0.0.0/${GIT_TAG:-0.0.0}/" package.json
```
Comment (austinmatherne-wk) [inline Dockerfile:19]: Can you drop the line number and instead search/replace the whole line (`"version": "0.0.0"`)?

File: setup.py [hunk 2]
```diff
@@ -16,7 +16,7 @@ def get_requirements():
 
 setup(
     name='ixbrl_viewer',
-    version='0.0.0',
+    version='@VERSION@',
```
Comment (brettkail-wk) [inline setup.py:19]: Why change the placeholder?  That seems unrelated to the issue described in the ticket, which is that the line number is specific.  (It might be good to include the relevant detail in the PR description given this is an open source repo.)

**Review by philipbalvanz-wf — DISMISSED** [review]

**Review by sagesmith-wf — APPROVED** [review]
No range pinning?

Won't let me delete but I saw other PR

File: Dockerfile [hunk 3]
```diff
@@ -13,14 +13,11 @@ RUN pip install -r requirements-dev.txt
...
+# The following command replaces the version string in setup.py and package.json
 # with the tagged version number from GIT_TAG or `0.0.0` if GIT_TAG is not set
 ARG VERSION=${GIT_TAG:-0.0.0}
-RUN echo "Version = $VERSION"
-# Update line 6 in setup.py
-RUN sed -i "6 s/0.0.0/$VERSION/" setup.py
-# Update line 3 in package.json
-RUN sed -i "3 s/0.0.0/$VERSION/" package.json
+RUN sed -i "s+version=\'0.0.0\'+version=\'$VERSION\'+" setup.py
+RUN sed -i "s+\"version\": \"0.0.0\"+\"version\": \"$VERSION\"+" package.json
```
Comment (brettkail-wk) [inline Dockerfile:20]: There's no need to escape `\'` inside `""`.  Pre-existing, the `.` are regex and probably should be escaped.  Out of curiosity, why change from conventional `/` to `+`?
```
RUN sed -i "s/version='0\.0\.0'/version='$VERSION'/" setup.py
RUN sed -i "s/\"version\": \"0\.0\.0\"/\"version\": \"$VERSION\"/" package.json
```
Well, I probably would have just kept the existing regexes and just removed the line number address (and possibly fixed the `.` escaping) since it seems unlikely that will match the wrong thing.

**Review by brettkail-wk — DISMISSED** [review]

**Review by sagesmith-wf — APPROVED** [review]

**Review by brettkail-wk — APPROVED** [review]

**Review by sagesmith-wf — APPROVED** [review]
QA +10

@Workiva/release-management-pp

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM
